From 4985b5001d4f2f64bbee7e9d6ee32058caf67252 Mon Sep 17 00:00:00 2001
From: Mark Andrews <marka@isc.org>
Date: Fri, 1 Sep 2017 11:17:59 +1000
Subject: [PATCH] 4697.   [bug]           Restore workaround for Microsoft
 Windows TSIG hash                         computation bug. [RT #45854]

(cherry picked from commit a8a20462b516b0cc39e9b1fb1a8dd514eb1aed29)
(cherry picked from commit b301c4293c082fcce4ec26218e6fad346976eb9e)
---
 CHANGES            | 3 +++
 lib/dns/rdataset.c | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/CHANGES b/CHANGES
index 5aa505345c..13b60473b5 100644
--- a/CHANGES
+++ b/CHANGES
@@ -1,3 +1,6 @@
+4697.	[bug]		Restore workaround for Microsoft Windows TSIG hash
+			computation bug. [RT #45854]
+
 	--- 9.11.2-P1 released ---
 
 4858.	[security]	Addresses could be referenced after being freed
diff --git a/lib/dns/rdataset.c b/lib/dns/rdataset.c
index a8e75d6caf..7eb394c8c4 100644
--- a/lib/dns/rdataset.c
+++ b/lib/dns/rdataset.c
@@ -467,6 +467,9 @@ towiresorted(dns_rdataset_t *rdataset, const dns_name_t *owner_name,
 	dns_name_copy(owner_name, name, NULL);
 	dns_rdataset_getownercase(rdataset, name);
 
+	if ((owner_name->attributes & DNS_NAMEATTR_NOCOMPRESS) != 0)
+		name->attributes |= DNS_NAMEATTR_NOCOMPRESS;
+
 	do {
 		/*
 		 * Copy out the name, type, class, ttl.
-- 
2.16.3

